### ISSUE TITLE
Build error when importing SnapKit in macOS project

### ISSUE THREAD

**Issue by JamieWhite** [SnapKit/SnapKit#346]
### New Issue Checklist

* [ ✓] I have looked at the [Documentation](http://snapkit.io/docs)
* [ ✓] I have read the [F.A.Q.](http://snapkit.io/faq)
* [ ✓] I have filled out this issue template.

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform                | macOS
 Platform Version        | 10.12
 SnapKit Version         | 3.0.1
 Integration Method      | carthage
 

### Issue Description

So this worked for me until I upgraded to the Swift 3.0 release. I'm guessing in your refactor you broke something? I notice in the project file you're using 1 target for multiple platforms. Perhaps this is the issue? I can't see a macOS deployment setting in the info tab where-as I see this in other frameworks that use single targets.

Here's a screenshot of my build error:
![Screenshot](https://dl.dropboxusercontent.com/s/2mbqalxirz26mnh/Screenshot%202016-10-04%2017.54.54.png?dl=0)

It obviously shouldn't be trying to import UIKit.

I've checked all the imports for UIKit and they're all guarded so they shouldn't be imported. I'm guessing there's something wrong with the Carthage archive.

I can't actually figure out what's wrong. The single target/multi platform thing is still new to me.

**Comment by JamieWhite** [comment]
So this is interesting. I've created a new project and tried running Carthage and I got this

```
jamie@Jamess-MacBook-Pro TestCarthageSnapkit $ carthage update --platform macos --no-use-binaries
*** Fetching SnapKit
*** Checking out SnapKit at "3.0.1"
*** xcodebuild output can be found in /var/folders/lc/hm7fx4dn4ls7plh80py2t6pr0000gn/T/carthage-xcodebuild.vgCmDA.log
*** Building scheme "SnapKit" in SnapKit.xcworkspace
Failed to write to /Users/jamie/Documents/code/TestCarthageSnapkit/Carthage/Build/Mac/SnapKit.framework: Error Domain=NSCocoaErrorDomain Code=260 "The file “SnapKit.framework” couldn’t be opened because there is no such file." UserInfo={NSFilePath=/Users/jamie/Library/Developer/Xcode/DerivedData/SnapKit-dgztqultserzvkaqxpadafoudtwh/Build/Products/Release-iphoneos/SnapKit.framework, NSUnderlyingError=0x7ff1df4253e0 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
```

**Comment by aalenliang** [comment]
I got the same issue, and i thought it's some thing about Carthage and the project settings.
Carthage will try copy the framework in base sdk (/Build/Products/Release-iphoneos/SnapKit.framework) to /Build/Mac, which caused all the wired behavior (UIKit not found and No such file.)

I open an issue in carthage, and the team said it is due to project's wrong configuration.
https://github.com/Carthage/Carthage/issues/1513
However I tried the solution, it still doesn't work.

**I had a temporarily solution, tested for my macOS project:**
1. Open the Xcode workspace in Carthage/Checkout/SnapKit
2. Change the build settings for SnapKit & SnapKitTests
![screen shot 2016-10-05 at 1 41 36 pm](https://cloud.githubusercontent.com/assets/20056005/19102279/87d389b0-8b01-11e6-8bc3-6ad3c265fb3d.png)
![screen shot 2016-10-05 at 1 41 42 pm](https://cloud.githubusercontent.com/assets/20056005/19102280/8bfe6690-8b01-11e6-88ae-a375cac30f73.png)
3. Run `carthage build SnapKit`

and everything makes sense.

I'm using https://github.com/danielgindi/Charts, which has one target for all platforms, it has the same configuration as above, and it works fine for my macOS project.

**Comment by robertjpayne** [comment]
I wonder if this is related to how I configured SnapKit as a universal framework. I'll dig in further!

**Comment by robertjpayne** [comment]
Fixed with [3.0.2](https://github.com/SnapKit/SnapKit/releases/tag/3.0.2)

**Comment by JamieWhite** [comment]
Marvellous, thanks Robert. 🍻
